cmd/minishift/cmd/stop.go

```diff
@@ -61,12 +61,16 @@ func runStop(cmd *cobra.Command, args []string) {
 
 	fmt.Println("Stopping local OpenShift cluster...")
 
-	if !skipUnRegistration {
-		if _, err := cluster.UnRegister(api); err != nil {
+	if !skipUnRegistration && minishiftConfig.InstanceConfig.IsRegistered {
+		supportUnregistration, err := cluster.UnRegister(api)
```

Comment:
_supportUnregistration_  is an unfortunate name imo. It implies to me that whatever you call "checks" whether unregistration is supported. Judging by name, I would not expect that unregistration actually occurs. However, this is the case. Why not 'unregistered' or 'unregistrationSuccessful'. This also makes clear why we then clean up the instance config state.